tests/python/gstaichi/lang/fast_caching/test_src_hasher.py

```diff
@@ -133,3 +133,50 @@ def get_fileinfos(functions: list[Callable]) -> list[_wrap_inspect.FunctionSourc
     assert src_hasher.validate_cache_key(cache_key)
 
     assert not src_hasher.validate_cache_key("abcdefg")
+
+
+# Should be enough to run these on cpu I think, and anything involving
+# stdout/stderr capture is fairly flaky on other arch
+@test_utils.test(arch=ti.cpu)
+@pytest.mark.parametrize(
+    "print_non_pure",
+    [
+        None,
+        False,
+        True,
+    ],
+)
```

Comment:
One line would be fine.

Reply:
I personally prefer parametrize tests one per line. Makes it easy for my brain to parse.